assets/js/googlesitekit/datastore/user/authentication.js

```diff
@@ -0,0 +1,199 @@
+
+export const INITIAL_STATE = {
+	authentication: undefined,
+	isFetchingAuthentication: false,
+};
+
+export const actions = {
+	*fetchAuthentication() {
+		let response, error;
+
+		yield {
+			payload: {},
+			type: START_FETCH_AUTHENTICATION,
+		};
+
+		try {
+			response = yield {
+				payload: {},
+				type: FETCH_AUTHENTICATION,
+			};
+
+			yield actions.receiveAuthentication( response );
+
+			yield {
+				payload: {},
+				type: FINISH_FETCH_AUTHENTICATION,
+			};
+		} catch ( e ) {
+			error = e;
+			yield {
+				payload: { error },
+				type: CATCH_FETCH_AUTHENTICATION,
+			};
+		}
+
+		return { response, error };
+	},
+	/**
+	 * Stores connection info received from the REST API.
+	 *
+	 * @since 1.5.0
+	 * @private
+	 * @param {Object} authentication Authentication info from the API.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveAuthentication( authentication ) {
+		invariant( authentication, 'authentication is required.' );
+
+		return {
+			payload: { authentication },
+			type: RECEIVE_AUTHENTICATION,
+		};
+	},
+};
+
+export const controls = {
+	[ FETCH_AUTHENTICATION ]: () => {
+		return API.get( 'core', 'user', 'authentication' );
+	},
+};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case START_FETCH_AUTHENTICATION: {
+			return {
+				...state,
+				isFetchingAuthentication: true,
+			};
+		}
+
+		case RECEIVE_AUTHENTICATION: {
+			const { authentication } = payload;
+
+			return {
+				...state,
+				authentication,
+			};
+		}
+
+		case FINISH_FETCH_AUTHENTICATION: {
+			return {
+				...state,
+				isFetchingAuthentication: false,
+			};
+		}
+
+		case CATCH_FETCH_AUTHENTICATION: {
+			return {
+				...state,
+				error: payload.error,
+				isFetchingAuthentication: false,
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+export const selectors = {
+
+	/**
+	 * Gets the authentication info for this user.
+	 *
+	 * Returns `undefined` if the authentication info is not available/loaded.
+	 *
+	 * Returns an object with the shape when successful:
+	 * ```
+	 * {
+	 *   authenticated: <Boolean>,
+	 * }
+	 * ```
+	 *
+	 * @private
+	 * @since 1.8.0
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object|undefined} User authentication info.
+	 */
+	getAuthentication( state ) {
+		const { authentication } = state;
+		return authentication;
+	},
+	/**
+	 * Gets the Site Kit authentication status for this user.
+	 *
+	 * Returns `true` if the site is connected to Site Kit, `false` if
+	 * not. Returns `undefined` if the connection info is not available/loaded.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean|undefined} Site connection status.
+	 */
+	isAuthenticated: createRegistrySelector( ( select ) => () => {
+		const authentication = select( STORE_NAME ).getAuthentication();
+
+		return typeof authentication !== 'undefined' ? authentication.authenticated : authentication;
```

Comment:
We have mixed use in the codebase, but let's standardize on `var === undefined` rather than using `typeof`